src/BigInt.elm

```diff
@@ -466,6 +488,48 @@ revMagnitudeToString (Magnitude digits) =
             String.concat <| Basics.toString x :: List.map fillZeroes xs
 
 
+{-| Print the number as a hex string.
+-}
+toHexString : BigInt -> String
+toHexString bigInt =
+    case bigInt of
+        Zer ->
+            "0"
+
+        Pos mag ->
+            if mag == Magnitude [] then
+                "0"
+            else
+                hexMagnitudeToString (Pos mag)
+
+        Neg mag ->
+            "-" ++ (toHexString (mul (fromInt -1) bigInt))
+
+-- Shortcut conversion to int for hex handling
+bigIntToInt_ : BigInt -> Int
+bigIntToInt_ bigInt =
+    case bigInt of
+        Zer -> 0
+
+        Pos (Magnitude [a]) -> a
+               
+        Pos (Magnitude [a,b]) -> b * (10^maxDigitMagnitude) + a
+
+        _ -> Debug.crash "No suitable shortcut conversion in hexMagnitudeToString"
+                
+hexMagnitudeToString : BigInt -> String
+hexMagnitudeToString bigInt =
+    case divmod bigInt eightHexDigits of
+        Nothing ->
+            Debug.crash "Failure converting to hex string."
```

Comment:
Is `Debug.crash` necessary?

Reply:
divmod doesn't know that eightHexDigits is a constant that can't be zero, and this is an internal function used by hex conversion.

The crash should be unreachable but should give an appropriate diagnostic if something goes really wrong.  Returning a failure didn't seem appropriate because an error here definitely wouldn't be the user's fault, but it could definitely go that route.

I'm open to suggestions if you have a different preference for this kind of thing.